static/assets/js/entity-reference/app.1.2.js

```diff
@@ -27,24 +27,39 @@
         function onInit() {
             bbWait.beginPageWait({});
 
-            $http.get(this.swaggerUrl)
-                .then(handleSuccess.bind(this), handleError.bind(this))
-                .finally(function() { bbWait.endPageWait(); });
+            var swaggerResponseCache = localStorageService.get('swaggerResponseCache');
+            if (!swaggerResponseCache || Math.abs(new Date(swaggerResponseCache.lastUpdatedDate).getTime() - new Date().getTime()) / 36e5 >= 12) {
+              $http.get(this.swaggerUrl)
+                  .then(handleSuccess.bind(this), handleError.bind(this))
+                  .finally(function() { bbWait.endPageWait(); });
+            }
+            else {
+              handleSwaggerResponse.bind(this)(swaggerResponseCache.response);
+              bbWait.endPageWait();
+            }
         }
 
         function handleSuccess(response) {
+            localStorageService.set('swaggerResponseCache', {
+              'response': response,
+              'lastUpdatedDate': new Date()
```

Comment:
Might be easier to have "expirationDate" = new Date() + 12hours & then above check if now > expirationDate in onInit